Bio/SeqIO/SnapGeneIO.py

```diff
@@ -0,0 +1,255 @@
+            for value in qualifier.getElementsByTagName('V'):
+                if value.hasAttribute('text'):
+                    qvalues.append(_decode(value.attributes['text'].value))
+                elif value.hasAttribute('predef'):
+                    qvalues.append(_decode(value.attributes['predef'].value))
+                elif value.hasAttribute('int'):
+                    qvalues.append(int(value.attributes['int'].value))
+            quals[qname] = qvalues
+
+        feature = SeqFeature(location, type=type, qualifiers=quals)
+        record.features.append(feature)
+
+
+def _parse_primers_packet(length, data, record):
+    """Parse a Primers packet.
+
+    A Primers packet is similar to a Features packet but specifically
+    stores primer binding features. The data is a XML string starting
+    with a 'Primers' root node.
+    """
+    xml = parseString(data.decode('ASCII'))
+    for primer in xml.getElementsByTagName('Primer'):
+        quals = {}
+
+        name = _get_attribute_value(primer, 'name')
+        if name:
+            quals['label'] = [name]
+
+        for site in primer.getElementsByTagName('BindingSite'):
+            rng = _get_attribute_value(site, 'location', error="Missing binding site location")
+            start, end = [int(x) for x in rng.split('-')]
+
+            strand = int(_get_attribute_value(site, 'boundStrand', default="0"))
+            if strand == 1:
+                strand = -1
+            else:
+                strand = +1
+
+            feature = SeqFeature(FeatureLocation(start, end, strand=strand), type='primer_bind', qualifiers=quals)
+            record.features.append(feature)
+
+
+_packet_handlers = {
+    0x00: _parse_dna_packet,
+    0x05: _parse_primers_packet,
+    0x06: _parse_notes_packet,
+    0x09: _parse_cookie_packet,
+    0x0A: _parse_features_packet
+    }
+
+
+# Helper functions to process the XML data in
+# some of the segments
+
+def _decode(text):
+    # Get rid of HTML tags in some values
+    return sub('<[^>]+>', '', text)
+
+
+def _get_attribute_value(node, name, default=None, error=None):
+    if node.hasAttribute(name):
+        return _decode(node.attributes[name].value)
+    elif error:
+        raise ValueError(error)
+    else:
+        return default
+
+
+def _get_child_value(node, name, default=None, error=None):
+    children = node.getElementsByTagName(name)
+    if children and children[0].childNodes and children[0].firstChild.nodeType == node.TEXT_NODE:
+        return _decode(children[0].firstChild.data)
+    elif error:
+        raise ValueError(error)
+    else:
+        return default
+
+
+def SnapGeneIterator(handle):
+    """Parse a SnapGene file and return a SeqRecord object."""
+    record = SeqRecord(None)
+    n = 0
+
+    for type, length, data in _PacketIterator(handle):
+        if n == 0 and type != 0x09:
+            raise ValueError("The file does not start with a SnapGene cookie packet")
+
+        if type in _packet_handlers:
```

Comment:
Are unknown packets comment or rare? If rare, does a ``BiopythonParserWarning`` seem appropriate?

Reply:
No, they are a common occurrence. In fact all the SnapGene files I have contain at least two or three unknown packets. Therefore I don’t think a warning would be appropriate here.

Reply:
Agreed.